store/src/java/com/zimbra/cs/mailbox/Tag.java

```diff
@@ -57,7 +57,9 @@
         }
 
         NormalizedTags(Mailbox mbox, String[] tagsFromClient, boolean create, boolean imapVisible) throws ServiceException {
-            assert mbox.isTransactionActive() : "cannot instantiate NormalizedTags outside of a transaction";
```

Comment:
Why did you comment this out? `isTransactionActive` seems like a valid reference.

https://github.com/Zimbra/zm-mailbox/blob/cd82ccc06afb04603e8fb9a45b5983a9b4773313/store/src/java/com/zimbra/cs/mailbox/Mailbox.java#L9981

Reply:
isTransactionActive() method is not accesible like a direct method for mailbox reference, you need to pass for a MailboxTransaction reference, in anycase, how can you know isTransactionActive is called for what reference?

Reply:
I just committed a small change that moves `isTransactionActive` back up to the `Mailbox` level. That should keep this code working.

Good catch!